cmd/hatchet-migrate/migrate/migrations/20251215195000_v1_batching_consolidated.sql

```diff
@@ -0,0 +1,1419 @@
+-- +goose Up
+-- +goose StatementBegin
+-- v0 schema alignment
+ALTER TYPE "LeaseKind" ADD VALUE IF NOT EXISTS 'BATCH';
+
+-- v0 "Step" batching configuration fields
+ALTER TABLE "Step"
+    ADD COLUMN IF NOT EXISTS batch_size INTEGER,
+    ADD COLUMN IF NOT EXISTS batch_flush_interval_ms INTEGER,
+    ADD COLUMN IF NOT EXISTS batch_key_expression TEXT,
+    ADD COLUMN IF NOT EXISTS batch_max_runs INTEGER;
+
+-- v1 batching propagation fields
+ALTER TABLE v1_task
+    ADD COLUMN IF NOT EXISTS batch_key TEXT;
+
+ALTER TABLE v1_queue_item
+    ADD COLUMN IF NOT EXISTS batch_key TEXT;
+
+ALTER TABLE v1_rate_limited_queue_items
+    ADD COLUMN IF NOT EXISTS batch_key TEXT;
+
+ALTER TABLE v1_task_runtime
+    ADD COLUMN IF NOT EXISTS batch_id UUID,
+    ADD COLUMN IF NOT EXISTS batch_size INTEGER,
+    ADD COLUMN IF NOT EXISTS batch_index INTEGER,
+    ADD COLUMN IF NOT EXISTS batch_key TEXT;
+
+CREATE INDEX IF NOT EXISTS v1_task_runtime_batch_id_idx
+    ON v1_task_runtime USING BTREE (batch_id)
+    WHERE batch_id IS NOT NULL;
+
+-- Batched queue items buffer table
+CREATE TABLE IF NOT EXISTS v1_batched_queue_item (
+    id BIGINT GENERATED ALWAYS AS IDENTITY,
+    tenant_id UUID NOT NULL,
+    queue TEXT NOT NULL,
+    task_id BIGINT NOT NULL,
+    task_inserted_at TIMESTAMPTZ NOT NULL,
+    external_id UUID NOT NULL,
+    action_id TEXT NOT NULL,
+    step_id UUID NOT NULL,
+    workflow_id UUID NOT NULL,
+    workflow_run_id UUID NOT NULL,
+    schedule_timeout_at TIMESTAMP(3),
```

Comment:
well, this is frustrating - it does look like all of our `schedule_timeout_at` fields are this data type, no-op here but we should fix this at some point